### PR TITLE
fix: prevent course info bar overflow

### LINKED ISSUE #729: Prevent Text-Wrap in SectionTableBody and the Department Card Component
Screenshot 2023-10-12 at 11 09 23 AM

):< Long course names make the buttons resize and shifts the UI

The fix should truncate the course names whenever they overflow some predefined min-length of the course info button. This also happens to the School/Dept card.

### PR DESCRIPTION
**Pull request by andrew-wang0**
## Summary

Prevent overflow

## Issues

Closes #729 

## Future Followup

will not be addressing

### CODE REVIEW

File: apps/antalmanac/src/components/RightPane/SectionTable/CourseInfoButton.tsx [hunk 1]
```diff
@@ -65,11 +63,13 @@ function CourseInfoButton({
         }
     };
 
+    const scheduleManagementWidth = useScheduleManagementStore((state) => state.scheduleManagementWidth);
+    const compact = scheduleManagementWidth && scheduleManagementWidth < 600;
```
Comment (KevinWu098) [inline apps/antalmanac/src/components/RightPane/SectionTable/CourseInfoButton.tsx]: issue: we should keep our breakpoints standard. prefer using a specific breakpoint as defined in our theme (and referencing it as such, not as a magic number)

File: apps/antalmanac/src/components/RightPane/SectionTable/CourseInfoButton.tsx [hunk 2]
```diff
@@ -87,7 +87,20 @@ function CourseInfoButton({
                     }
                 }}
             >
-                {text}
+                <span style={{ display: 'flex', gap: 4 }}>
+                    {icon}
```
Comment (KevinWu098) [inline apps/antalmanac/src/components/RightPane/SectionTable/CourseInfoButton.tsx:96]: question/issue: can we continue passing in icon as `startIcon`? Why are we moving this into the children?

**Review by KevinWu098 — CHANGES_REQUESTED** [review]
Thanks for this PR! Really looking forward to the UI improvement via reduction of UI shift. 

Have a couple comments/nits to resolve, but otherwise, looks solid!

**Review by andrew-wang0 — COMMENTED** [review]
🫡

File: apps/antalmanac/src/components/RightPane/SectionTable/CourseInfoButton.tsx (hunk 2 above)
Comment (andrew-wang0) [reply apps/antalmanac/src/components/RightPane/SectionTable/CourseInfoButton.tsx:96]: startIcon auto adds margin right, making it not centered

**Review by alexanderl19 — APPROVED** [review]

**Review by KevinWu098 — APPROVED** [review]

**Pull request merged**